### PR TITLE
US4.3 Generation of all permutation for given Board size

### PR DESCRIPTION
**Pull request by jzych**

### CODE REVIEW

**Comment by LordLukin** [comment]
It looks that this PR includes changes from the other one. They should be done separately. If it required previous changes it means that you shouldn't have started this task before merging the previous one.

**Comment by jzych** [comment]
Branch fixed, now only contains commits for this issue.

File: test/GeneratorTests.cpp [hunk 1]
```diff
@@ -23,3 +16,21 @@ TEST_F(GeneratorTests, checkConversionFromIntToFieldsEnum)
...
+    ASSERT_EQ(s.size(), pow(2,(rows * cols)));
+}
+
+TEST_F(GeneratorTests, checkSizeOfSAllPermutationOfBoard4x3)
+{
+    int rows = 4;
+    int cols = 3;
+    auto result = GenerateAllBoardPermutations(rows,cols);
+    std::set<Lines> s(result.begin(), result.end());
+    ASSERT_EQ(s.size(), pow(2,(rows * cols)));
```
Comment (LordLukin) [inline test/GeneratorTests.cpp]: It's the same test as a previous one, only values for rows and cols differ. You can use TEST_P to write test case only once and check it on different inputs

Comment (jzych) [reply test/GeneratorTests.cpp]: Added parameterized test case.

File: test/GeneratorTests.cpp [hunk 2]
```diff
@@ -23,3 +20,19 @@ TEST_F(GeneratorTests, checkConversionFromIntToFieldsEnum)
 
     ASSERT_EQ(result, ConvertIntToFieldEnum(input));
 }
+
+INSTANTIATE_TEST_CASE_P(Default,
+                        GeneratorParamTests,
+                        testing::Combine(testing::Range(3, 5),   // rows
```
Comment (LordLukin) [inline test/GeneratorTests.cpp]: These comments are misleading. 3 is row, 5 is cols. Please remove comments and you can also add some more cases like 2, 2 or 5, 5

Comment (jzych) [reply test/GeneratorTests.cpp]: In this part there are passed tuples<int, int> to board constructor. There ale genereted all cases for board col/row size from 3 to 4. Board of size 5x5 has 2^25 permutations and this is more than my computer has memory. Board with col/rows smaller then 3 are found to be invalid in JSONFileREader.

File: src/Generator.cpp [hunk 3]
```diff
@@ -1,25 +1,18 @@
 #include "Generator.hpp"
 
-Lines getAllVectorPermutation_(int vSize)
+bool nextPermutation(Line &row)
```
Comment (gitmajo) [inline src/Generator.cpp]: Please keep existing style where &, * is sticked to Type. 
Line& row <--- 
this convention is used on master in Board.cpp

Please change it also below.

Comment (jzych) [reply src/Generator.cpp]: done

**Pull request merged**